vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryCollectionOperatorDescriptor.java

```diff
@@ -129,11 +137,26 @@ public void nextFrame(ByteBuffer buffer) throws HyracksDataException {
                                 Iterator<File> it = FileUtils.iterateFiles(collectionDirectory,
                                         new VXQueryIOFileFilter(), TrueFileFilter.INSTANCE);
                                 while (it.hasNext()) {
-                                    File xmlDocument = it.next();
-                                    if (LOGGER.isLoggable(Level.FINE)) {
-                                        LOGGER.fine("Starting to read XML document: " + xmlDocument.getAbsolutePath());
+                                    File file = it.next();
+                                    if (file.getName().toLowerCase().endsWith(".xml")) {
```

Comment:
Can we pull the `file.getName().toLowerCase()` into a variable, so that it won't be evaluated twice?